cpp/src/Slice/Grammar.y

```diff
@@ -677,7 +677,7 @@ optional_type_id
     // It's safe to perform this check in the parser, since we already have enough information to know whether a type
     // can be optional. This is because the only types that can be forward declared (classes/interfaces) have constant
     // values for `usesClasses` (true/false respectively).
-    if (m->type->usesClasses())
+    if (m->type && m->type->usesClasses())
```

Comment:
There were a handful of places where I had to add checks for `nullptr` before using a `TypePtr` now.